resources/lib/KodiHelper.py

```diff
@@ -302,11 +307,66 @@ def get_credentials (self):
         :obj:`dict` of :obj:`str`
             The users stored account data
         """
+        addon = self.get_addon()
+        email = addon.getSetting('email')
+        password = addon.getSetting('password')
+
+        # soft migration for existing credentials
+        # base64 can't contain `@` chars
+        if '@' in email:
+            addon.setSetting('email', self.encode(raw=email))
+            addon.setSetting('password', self.encode(raw=password))
+            return {
+                'email': self.get_addon().getSetting('email'),
+                'password': self.get_addon().getSetting('password')
+            }
+        
+        # if everything is fine, we decode the values
+        if '' != email or '' != password:
+            return {
+                'email': self.decode(enc=email),
+                'password': self.decode(enc=password)
+            }
+
+        # if email is empty, we return an empty map
         return {
-            'email': self.get_addon().getSetting('email'),
-            'password': self.get_addon().getSetting('password')
+            'email': '',
+            'password': ''                
         }
 
+    def encode(self, raw):
+        """
+        Encodes data
+
+        :param data: Data to be encoded
+        :type data: str
+        :returns:  string -- Encoded data
+        """
+        raw = self._pad(raw)
+        iv = Random.new().read(AES.block_size)
+        cipher = AES.new(self.crypt_key, AES.MODE_CBC, iv)
+        return base64.b64encode(iv + cipher.encrypt(raw))
+
+    def decode(self, enc):
+        """
+        Decodes data
+
+        :param data: Data to be decoded
+        :type data: str
+        :returns:  string -- Decoded data
+        """
+        enc = base64.b64decode(enc)
+        iv = enc[:AES.block_size]
+        cipher = AES.new(self.crypt_key, AES.MODE_CBC, iv)
+        return self._unpad(cipher.decrypt(enc[AES.block_size:])).decode('utf-8')
+
+    def _pad(self, s):
```

Comment:
Isn't it possible to use Crypto.Util.Padding ?

Reply:
I haven't checked as I didn't know about `Crypto.Util.Padding`.
Do you think it might have a speed benefit of a measurable margin?

Reply:
Dont think that it has speed benefit but it it would remove unnecessary code

Reply:
```
from Crypto.Util import Padding
ImportError: cannot import name Padding
```

Not sure why it cant... ... ...so I would roll with the current impl. & may investigate improvements when we´re refactoring things.

Reply:
I agree with asciidisco - didn't find a solution at the moment, too.

Reply:
Cryptodome instead of crypto because we're using pycryptodomex

Reply:
@trummerjo your are right - this fixed the error